src/plugins/discover/public/application/utils/state_management/discover_slice.tsx

```diff
@@ -39,36 +40,63 @@ export interface DiscoverRootState extends RootState {
   discover: DiscoverState;
 }
 
-const initialState = {} as DiscoverState;
+const initialState: DiscoverState = {
+  columns: [],
```

Comment:
We should add `_source` to initial columns state
```
const initialState: DiscoverState = {
  columns: ['_source'],
  sort: [],
};
```
Otherwise we will see only time column when first render discover table
<img width="839" alt="Screenshot 2023-08-03 at 11 05 48" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/f461839a-8325-4bc4-8856-1b4c15e56489">